Content.Server/_Harmony/JoinQueue/JoinQueueManager.cs

```diff
@@ -0,0 +1,169 @@
+using System.Linq;
+using Content.Server.Connection;
+using Content.Shared.CCVar;
+using Content.Shared._Harmony.Common.JoinQueue;
+using Prometheus;
+using Robust.Server.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Enums;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using Content.Shared._Harmony.CCVars;
+
+namespace Content.Server._Harmony.JoinQueue;
+
+/// <summary>
+///     Manages new player connections when the server is full and queues them up, granting access when a slot becomes free
+/// </summary>
+public sealed class JoinQueueManager : IJoinQueueManager
+{
+    private static readonly Gauge QueueCount = Metrics.CreateGauge(
+        "join_queue_total_count",
+        "Amount of players in queue.");
+
+    private static readonly Histogram QueueTimings = Metrics.CreateHistogram(
+        "join_queue_timings",
+        "Timings of players in queue",
+        new HistogramConfiguration()
+        {
+            LabelNames = new[] { "type" },
+            Buckets = Histogram.ExponentialBuckets(1, 2, 14),
+        });
+
+
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly IConfigurationManager _configuration = default!;
+    [Dependency] private readonly IServerNetManager _net = default!;
+    [Dependency] private readonly IConnectionManager _connection = default!;
+
+    /// <summary>
+    /// Queue of active player sessions
+    /// </summary>
+    private readonly List<ICommonSession> _queue = new();
+
+    private bool _isEnabled = false;
```

Comment:
The default for a boolean is already false
```suggestion
    private bool _isEnabled;
```